lib/frame.js

```diff
@@ -0,0 +1,25 @@
+class Frame {
```

Comment:
I love that you have a separate Frame class, really good distribution of responsibilities.